### ISSUE TITLE
Channel Updates Issue

### ISSUE THREAD

**Issue by barkerja** [Kraigie/nostrum#87]
Is there a known issue with the `CHANNEL_UPDATE` event when it's triggered by a permission overwrite? The consumer crashes when this occurs:

```
15:34:05.436 [error] GenServer #PID<0.437.0> terminating
** (stop) exited in: GenServer.call(#PID<0.435.0>, {:update, :channel, %Nostrum.Struct.Channel{application_id: nil, bitrate: nil, guild_id: 105713369939480576, icon: nil, id: 419255918199439363, last_message_id: 458680159143460864, last_pin_timestamp: nil, name: "bot-testing", nsfw: false, owner_id: nil, parent_id: nil, permission_overwrites: [%Nostrum.Struct.Overwrite{allow: 0, deny: 1024, id: 105713369939480576, name: nil}, %Nostrum.Struct.Overwrite{allow: 805829713, deny: 0, id: 168192499293224960, name: nil}], position: 24, recipients: nil, topic: nil, type: 0, user_limit: nil}}, 5000)
    ** (EXIT) an exception was raised:
        ** (Protocol.UndefinedError) protocol Enumerable not implemented for %Nostrum.Struct.Overwrite{allow: 0, deny: 1024, id: 105713369939480576, name: nil}. This protocol is implemented for: DBConnection.PrepareStream, DBConnection.Stream, Date.Range, File.Stream, Function, GenEvent.Stream, HashDict, HashSet, IO.Stream, List, Map, MapSet, Postgrex.Stream, Range, Stream
            (elixir) /private/tmp/elixir-20180316-64850-zsrybb/elixir-1.6.4/lib/elixir/lib/enum.ex:1: Enumerable.impl_for!/1
            (elixir) /private/tmp/elixir-20180316-64850-zsrybb/elixir-1.6.4/lib/elixir/lib/enum.ex:141: Enumerable.reduce/3
            (elixir) lib/enum.ex:1911: Enum.reverse/1
            (elixir) lib/enum.ex:2588: Enum.to_list/1
            (elixir) lib/map.ex:192: Map.new/2
            (nostrum) lib/nostrum/struct/overwrite.ex:44: Nostrum.Struct.Overwrite.to_struct/1
            (elixir) lib/enum.ex:1294: Enum."-map/2-lists^map/1-0-"/2
            (elixir) lib/map.ex:571: Map.update/4
Last message: {:DOWN, #Reference<0.1399560593.7340038.203953>, :process, #PID<0.436.0>, {{%Protocol.UndefinedError{description: "", protocol: Enumerable, value: %Nostrum.Struct.Overwrite{allow: 0, deny: 1024, id: 105713369939480576, name: nil}}, [{Enumerable, :impl_for!, 1, [file: '/private/tmp/elixir-20180316-64850-zsrybb/elixir-1.6.4/lib/elixir/lib/enum.ex', line: 1]}, {Enumerable, :reduce, 3, [file: '/private/tmp/elixir-20180316-64850-zsrybb/elixir-1.6.4/lib/elixir/lib/enum.ex', line: 141]}, {Enum, :reverse, 1, [file: 'lib/enum.ex', line: 1911]}, {Enum, :to_list, 1, [file: 'lib/enum.ex', line: 2588]}, {Map, :new, 2, [file: 'lib/map.ex', line: 192]}, {Nostrum.Struct.Overwrite, :to_struct, 1, [file: 'lib/nostrum/struct/overwrite.ex', line: 44]}, {Enum, :"-map/2-lists^map/1-0-", 2, [file: 'lib/enum.ex', line: 1294]}, {Map, :update, 4, [file: 'lib/map.ex', line: 571]}]}, {GenServer, :call, [#PID<0.435.0>, {:update, :channel, %Nostrum.Struct.Channel{application_id: nil, bitrate: nil, guild_id: 105713369939480576, icon: nil, id: 419255918199439363, last_message_id: 458680159143460864, last_pin_timestamp: nil, name: "bot-testing", nsfw: false, owner_id: nil, parent_id: nil, permission_overwrites: [%Nostrum.Struct.Overwrite{allow: 0, deny: 1024, id: 105713369939480576, name: nil}, %Nostrum.Struct.Overwrite{allow: 805829713, deny: 0, id: 168192499293224960, name: nil}], position: 24, recipients: nil, topic: nil, type: 0, user_limit: nil}}, 5000]}}}
State: %ConsumerSupervisor{args: [UsmcBot.Handler, []], children: %{}, max_restarts: 3, max_seconds: 5, mod: Nostrum.Consumer, name: {#PID<0.437.0>, Nostrum.Consumer}, producers: %{}, restarting: 0, restarts: [], strategy: :one_for_one, template: {UsmcBot.Handler, {UsmcBot.Handler, :start_link, []}, :transient, 5000, :worker, [UsmcBot.Handler]}}
```

**Comment by barkerja** [comment]
This actually appears to happen for _all_ `CHANNEL_UPDATE` events, not just events triggered specifically by a permission overwrite. https://github.com/Kraigie/nostrum/commit/e9e5afd91a33f9c7b0cdbb03bbcc8de8b0ba0aca appears to be the offending commit.

**Comment by Kraigie** [comment]
This was fixed with 27979faed3f2d3b4d9593869155fb739554d91ad. Thanks for the report!
